sw/banshee/src/main.rs

```diff
@@ -215,7 +216,15 @@ fn main() -> Result<()> {
     }
 
     engine.config = if let Some(config_file) = matches.value_of("configuration") {
-        Configuration::parse(config_file, engine.num_clusters)
+        // if configuration file is given and `architecture` information is set
+        // use that configuration, else banshee parameter
+        let config_used: Configuration = Configuration::parse(config_file, engine.num_clusters);
+        if config_used.architecture.num_clusters != 0 {
+            engine.num_cores = config_used.architecture.num_cores;
+            engine.num_clusters = config_used.architecture.num_clusters;
+            engine.base_hartid = config_used.architecture.base_hartid;
+        }
+        config_used
```

Comment:
I agree that it would be nice to have the option to override the config file with command line arguments. Would moving this block above the `num-*` blocks be enough?

Reply:
it was not enough, but I implemented it now correctly.